CRM/Member/Form/Membership.php

```diff
@@ -1456,8 +1460,8 @@ public function submit() {
         }
       }
 
-      if ($formValues['payment_status_id'] != array_search('Completed', $allContributionStatus)) {
-        $params['status_id'] = array_search('Pending', $allMemberStatus);
+      if ($paymentStatus !== 'Completed') {
```

Comment:
@mattwire this PR is your code - except I added a few lines to the test that failed when re-running it & I altered this line - setting paymentStatus at the point it becomes available & using it here